test/new-e2e/tests/apm/tests.go

```diff
@@ -68,6 +70,57 @@ func testTracesHaveContainerTag(t *testing.T, c *assert.CollectT, service string
 	assert.True(c, hasContainerTag(traces, fmt.Sprintf("container_name:%s", service)))
 }
 
+func testAutoVersionTraces(t *testing.T, c *assert.CollectT, intake *components.FakeIntake) {
+	t.Helper()
+	traces, err := intake.Client().GetTraces()
+	assert.NoError(c, err)
+	assert.NotEmpty(c, traces)
+	t.Log("Got traces", traces)
+	for _, tr := range traces {
+		for _, tp := range tr.TracerPayloads {
+			t.Log("Tracer Payload Tags:", tp.Tags["_dd.tags.container"])
+			ctags, ok := getContainerTags(t, tp)
+			assert.True(t, ok)
+			imageID, ok := ctags["image_id"]
+			assert.True(t, ok)
+			t.Logf("Got image ID: %v", imageID)
+			assert.Equal(t, "sha256:76909dd636e2ac542b2ff22b134e55861807ec3b88117d3d8573da5adb5b22d0", imageID)
```

Comment:
The hash is expected to change when the `tracegen` image is rebuilt. We ca make assertions on the format (e.g the string starts with `sha256:` and the rest is a 256-bit hex).

Another alternative is to get the real image sha from docker directly, but in that case we would start testing the correctness of the tagger which is not in the scope of these tests.